### PR TITLE
[FLINK-26905] Re-add FlinkDeploymentList and FlinkSessionJobList classes

### PR DESCRIPTION
**Pull request by zeus1ammon**

### CODE REVIEW

**Review by Aitozi — COMMENTED** [review]
The file is absence of Apache license, Please take a look [here](https://nightlies.apache.org/flink/flink-docs-master/docs/flinkdev/ide_setup/#copyright-profile)

**Comment by mbalassi** [comment]
```
Error:  Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:2.17:check (validate) on project flink-kubernetes-operator: You have 2 Checkstyle violations. -> [Help 1]
```

**Comment by morhidi** [comment]
I'm not sure if we need these files, could you please elaborate? @zeus1ammon

**Comment by zeus1ammon** [comment]
This is needed for upstream projects, esp fabric8 java client in it is breaking without this. Otherwise, these classes has to be implemented on every of those Upstream projects separately, which will be a pain.

**Comment by mbalassi** [comment]
@zeus1ammon please add Javadoc comments to these classes describing why they are necessary, so folks do not accidentally remove them thinking that they are unused.

**Comment by zeus1ammon** [comment]
@mbalassi javadoc is added. please let me know, if the content of the javadoc is enough.

**Comment by mbalassi** [comment]
Thanks @zeus1ammon I will merge this latest tomorrow, might slightly modify wording of the comment. In principle I like it.

File: flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/crd/FlinkDeploymentList.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ */
+
+package org.apache.flink.kubernetes.operator.crd;
+
+import org.apache.flink.annotation.Experimental;
+
+import io.fabric8.kubernetes.client.CustomResourceList;
+
+/**
+ * Multiple Flink deployments. This class is needed by upstream projects, if they have to use this
```
Comment (tweise) [inline flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/crd/FlinkDeploymentList.java:25]: upstream => downstream

**Pull request closed without merging**